### ISSUE TITLE
Pipes (and presumably ducts, etc) export using system material not actual material

### ISSUE THREAD

**Issue by Moult** [Autodesk/revit-ifc#577]
If I click on a pipe and check its instance properties, I see a "Material" property with something like "PVC" or "Copper" set as a material. These are legit materials and found in the Materials browser / manager.

However, let's say these pipes are part of a system called "SEWER". Then, the exported IFC will show that the pipe has an IfcMaterial called "SEWER" assigned to it.

How do I get the pipe to export using its actual material, not the system name (which logically is not a material)?

**Comment by AngelVelezSosa** [comment]
They work that way because that was the overwhelming demand that we use the system material and not the actual material.  If there is use for both, that could be an option.  I'll look in the morning to see of I can find evidence of my claim.

**Comment by Moult** [comment]
Yeah probably because back in the day people got confused by styles (colors) and materials and wanted their color coded pipes and ducts to show up with the "right" color in IFC viewers. Its unfortunately semantically wrong: a pipe is made out of copper, not "hot water supply".

I think if you used the system material as the style, but the real material as the material, you can visually satisfy the old crowd and semantically deliver the correct data.

What do you think? Time to revisit this with an emphasis on correct data?

**Comment by CyrilWaechter** [comment]
@AngelVelezSosa I opened https://github.com/Autodesk/revit-ifc/issues/112 long time ago and as @Moult guessed how it is described was mainly result of a lack of knowledge about IFC. If I knew it was possible to get both (pipe actual material and system style) I would have written the issue this way. So I agree with @Moult it is far better to correct IFC export the way IFC is intended to be used and inform users and viewers developers about it rather than fixing in the wrong way.

**Comment by ArtemHnatko** [comment]
This issue is also reported in Jira as "REVIT-236144" and is currently in the "Open" status.

**Comment by CEM-EnriqueRuiz** [comment]
i am agree with @Moult opinion, i think current implementation is wrong.
